File: src/SwiftCart.Backend/SwiftCart.Domain/Entities/Order.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using SwiftCart.Domain.Enums;

namespace SwiftCart.Domain.Entities;

public class Order
{
    public Guid Id { get; set; }
    public string OrderNumber { get; set; } = string.Empty;
    public Guid UserId { get; set; }

    public string ShipToName { get; set; } = string.Empty;
    public string ShipStreet { get; set; } = string.Empty;
    public string ShipCity { get; set; } = string.Empty;
    public string ShipState { get; set; } = string.Empty;
    public string ShipCountry { get; set; } = string.Empty;
    public string ShipZipCode { get; set; } = string.Empty;

    public Guid? ShippingAddressId { get; set; }

    [Column(TypeName = "decimal(18, 2)")]
    public decimal SubTotal { get; set; }
    [Column(TypeName = "decimal(18, 2)")]
    public decimal ShippingCost { get; set; }
    [Column(TypeName = "decimal(18, 2)")]
    public decimal TaxAmount { get; set; }
    [Column(TypeName = "decimal(18, 2)")]
    public decimal TotalAmount { get; set; }

    public PaymentStatus PaymentStatus { get; set; }
    public OrderStatus OrderStatus { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public required User User { get; set; }
    public Address? ShippingAddress { get; set; }
    public List<OrderItem> Items { get; set; } = new();
}

File: src/SwiftCart.Client/src/app/ecommerce/models/order.ts
export enum OrderStatus {
  Pending = 'Pending',
  Confirmed = 'Confirmed',
  Packed = 'Packed',
  Shipped = 'Shipped',
  OutForDelivery = 'OutForDelivery',
  Delivered = 'Delivered',
  Canceled = 'Canceled',
  Returned = 'Returned'
}

export interface OrderItem {
	productId: string; // Guid as string
	productName: string;
  images?: string[];
	quantity: number;
	unitPrice: number;
 	lineTotal?: number;
}

export interface Order {
	id: string;
	orderNumber: string;
	userId: string;

	shipToName: string;
	shipStreet: string;
	shipCity: string;
	shipState: string;
	shipCountry: string;
	shipZipCode: string;

	subTotal: number;
	shippingCost: number;
	taxAmount: number;
	totalAmount: number;

	paymentStatus: string;
	orderStatus: string;

	createdAt: string;  

	items: OrderItem[];
}

export interface InitiateOrder{
 clientSecret:string;
 orderId:string;
 totalAmount:number;
 orderNumber:string;
}